pkg/image/api/types.go

```diff
@@ -296,6 +296,11 @@ type ImageStreamStatus struct {
 	// DockerImageRepository represents the effective location this stream may be accessed at. May be empty until the server
 	// determines where the repository is located
 	DockerImageRepository string
+	// PublicDockerImageRepository represents the public location from where the image can
+	// be pulled outside the cluster. The public URL is constructed by reading the route created for
+	// the docker-registry service. This field can be empty if the route is not created
+	// (iow. the integrated registry is not exposed)
+	PublicDockerImageRepository string
```

Comment:
@smarterclayton will tagging the external resolution policy cause the builds to push images via route? That does not seem right to me :-) We will have to generate the dockercfg secrets in that case for the route and watch the route for changes (and regenerate the secrets).

The `publicImageRepository` sounds ok to me.